File: frontend/mobile/lib/data/models/registration_request.dart
class RegistrationRequest {
  // User
  final String tenantKey;
  final String username;
  final String firstName;
  final String lastName;
  final String email;
  final String password;
  final List<String> roles;
  final UserProfile profile;

  RegistrationRequest({
    required this.tenantKey,
    required this.username,
    required this.firstName,
    required this.lastName,
    required this.email,
    required this.password,
    required this.roles,
    required this.profile,
  });

  Map<String, dynamic> toJson() {
    return {
      'tenantKey': "Farmer",
      'username': username,
      'firstName': firstName,
      'lastName': lastName,
      'email': email,
      'password': password,
      'roles': "Farmer",
      'profile': profile.toJson(),
    };
  }
}

class UserProfile {
  // User Profile
  final String rsbsaId;
  final String? middleName;
  final String phoneNumber;
  final String gender;
  final String civilStatus;
  final String street;
  final String barangay;
  final String municipality;
  final String province;
  final String region;
  final String farmerType;
  final double totalFarmAreaHa;

  UserProfile({
    required this.rsbsaId,
    this.middleName,
    required this.phoneNumber,
    required this.gender,
    required this.civilStatus,
    required this.street,
    required this.barangay,
    required this.municipality,
    required this.province,
    required this.region,
    required this.farmerType,
    required this.totalFarmAreaHa,
  });

  Map<String, dynamic> toJson() {
    return {
      'rsbsaId': rsbsaId,
      'middleName': middleName,
      'gender': gender,
      'civilStatus': civilStatus,
      'street': street,
      'barangay': barangay,
      'municipality': municipality,
      'province': province,
      'region': region,
      'farmerType': farmerType,
      'totalFarmAreaHa': totalFarmAreaHa,
    };
  }

}

File: services/farmer/src/main/java/com/hashjosh/farmer/dto/RegistrationRequest.java
package com.hashjosh.farmer.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;
import java.util.Set;
import java.util.UUID;

@Getter
@Setter
@Builder
public class RegistrationRequest {
    // User
    private String rsbsaId;
    private String password;
    private String firstName;
    private String lastName;
    private String middleName;
    private String email;
    private String phoneNumber;

    // User Profile
    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate dateOfBirth;
    private String gender;
    private String civilStatus;

    private String street;
    private String barangay;
    private String municipality;
    private String province;
    private String region;

    private String farmerType;
    private Double totalFarmAreaHa;
}
